processor/src/test/java/org/mapstruct/ap/test/targetthis/CustomerEntityMapper1.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.targetthis;
+
+import org.mapstruct.CollectionMappingStrategy;
+import org.mapstruct.InheritInverseConfiguration;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.ReportingPolicy;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(
+        unmappedSourcePolicy = ReportingPolicy.IGNORE,
+        unmappedTargetPolicy = ReportingPolicy.IGNORE,
+        collectionMappingStrategy = CollectionMappingStrategy.ADDER_PREFERRED
+)
+public interface CustomerEntityMapper1 {
+    CustomerEntityMapper1 INSTANCE = Mappers.getMapper( CustomerEntityMapper1.class );
+
+    @Mapping(target = ".", source = "entity")
+    CustomerEntity map(CustomerDTO customer);
+
+    @InheritInverseConfiguration(name = "map")
+    // @Mapping(target = "entity", source = ".")
+    CustomerDTO inverseMap(CustomerEntity customer);
+
+    @Mapping(target = "entity", source = ".")
+    void update(CustomerEntity customer, @MappingTarget CustomerDTO dto);
+
+    @Mapping(target = ".", source = "entity")
```

Comment:
could you also specify `customer` as target?

Reply:
probably not at this point, there is explicit check for target "." to read all sub-properties from source, if you use something else it should work exactly as before, so nothing is broken for others.

Reply:
> could you also specify `customer` as target?

I'll open up another issue for that.. I'm not sure if its possible already, but I think its not the in the scope of this PR.. (See #1705)